### PR TITLE
Partial fix, but usable-ish

### PR DESCRIPTION
**Pull request by md8n**
# Description

This matches the changes to the subscription API as described in the docs.

A separate :pr: will cover actually ensuring it works with Appelflap

### CODE REVIEW

File: src/ts/AppDataStatus.ts [hunk 1]
```diff
@@ -242,10 +242,12 @@ export class AppDataStatus {
         const published = await this.PublishAll();
         const unpublished = await this.UnpublishAll();
         let subscriptions = await this.GetSubscriptions();
-        let origins = Object.keys(subscriptions.origins);
+        let origins = Object.keys(subscriptions.types.CACHE.groups);
```
Comment (boerwastaken) [inline src/ts/AppDataStatus.ts:245]: There may not always be a `CACHE` key. Is that caught somewhere?
Note that any `map` type in [the relevant descriptor classes](https://github.com/catalpainternational/appelflap/blob/master/koekoek/src/main/java/org/nontrivialpursuit/ingeblikt/Descriptors.kt#L74:L92) may be empty.

File: src/ts/tests/AppelflapConnect.test.ts [hunk 2]
```diff
@@ -367,33 +367,37 @@ test.skip("Cache: setSubscriptions", async (t: any) => {
...
-                        p2p_version_max: 888,
-                        injected_version: 12,
-                    },
-                    "another-cache-name": {
-                        injection_version_min: 42,
-                        injection_version_max: 42,
-                        p2p_version_min: 1,
-                        p2p_version_max: 9000,
-                        injected_version: null,
+        types: {
```
Comment (boerwastaken) [inline src/ts/tests/AppelflapConnect.test.ts:370]: I'd recommend also throwing a SWORK type in here. And to test the empty map too.

**Review by boerwastaken — CHANGES_REQUESTED** [review]
Note that any `map` type in [the relevant descriptor classes](https://github.com/catalpainternational/appelflap/blob/master/koekoek/src/main/java/org/nontrivialpursuit/ingeblikt/Descriptors.kt#L74:L92) may be empty.

Are you sure that's reflected in your types and tests?

**Comment by PeteCoward** [comment]
I may regret asking, but what is [SWORK](https://www.urbandictionary.com/define.php?term=swork)?

**Comment by md8n** [comment]
> I may regret asking, but what is [SWORK](https://www.urbandictionary.com/define.php?term=swork)?

I'm guessing an abbreviation for Service Worker

File: src/ts/tests/AppelflapConnect.test.ts (hunk 2 above)
Comment (md8n) [reply src/ts/tests/AppelflapConnect.test.ts:401]: done

File: src/ts/AppDataStatus.ts (hunk 1 above)
Comment (md8n) [reply src/ts/AppDataStatus.ts]: done

**Comment by md8n** [comment]
> Are you sure that's reflected in your types and tests?

It is now

**Comment by boerwastaken** [comment]
> >  I may regret asking, but what is SWORK?

> I'm guessing an abbreviation for Service Worker

Youse _could_ grep the docs, ya know.

**Pull request merged**